src/relay/pass/type_infer.cc

```diff
@@ -363,14 +393,20 @@ class TypeInferencer : private ExprFunctor<Type(const Expr&)> {
     Array<Type> arg_types;
     for (size_t i = 0; i < f->params.size(); ++i) {
       Type atype = solver_.Resolve(GetType(f->params[i]));
-      CHECK(atype.as<IncompleteTypeNode>() == nullptr)
-          << "Cannot resolve type of " << i
-          << "-th parameter of function at" << f->span;
+      if (atype.as<IncompleteTypeNode>() != nullptr) {
+        this->ReportFatalError(
+          GetRef<Function>(f),
+          RELAY_ERROR("Cannot resolve type of ")
+          << i << "-th parameter of function");
```

Comment:
Do people expect this to be 0-indexed?

Reply:
I'm not sure, I wasn't super focused on changing the content of the error messages in this pass.